packages/api/src/folder/controller.test.ts

```diff
@@ -0,0 +1,311 @@
+  const agent = request(app);
+
+  beforeEach(async () => {
+    (verifyUserAuthentication as jest.Mock).mockImplementation(
+      async (
+        req: Request<
+          unknown,
+          unknown,
+          { userSubjectFromAuthToken?: string; emailFromAuthToken?: string }
+        >,
+        __,
+        next: NextFunction
+      ) => {
+        req.body.emailFromAuthToken = "test@permanent.org";
+        req.body.userSubjectFromAuthToken =
+          "b5461dc2-1eb0-450e-b710-fef7b2cafe1e";
+        next();
+      }
+    );
+    await clearDatabase();
+    await setupDatabase();
+  });
+
+  afterEach(async () => {
+    await clearDatabase();
+    jest.restoreAllMocks();
+    jest.clearAllMocks();
+  });
+
+  test("expect an empty query to cause a 400 error", async () => {
+    await agent.patch("/api/v2/folder/1").send({}).expect(400);
+  });
+
+  test("expect non existent folder to cause a 404 error", async () => {
+    await agent
+      .patch("/api/v2/folder/111111111")
+      .send({ displayDate: "2024-09-26T15:09:52.000Z" })
+      .expect(404);
+  });
+
+  test("expect request to have an email from auth token if an auth token exists", async () => {
+    (extractUserEmailFromAuthToken as jest.Mock).mockImplementation(
+      (req: Request, __, next: NextFunction) => {
+        (req.body as { emailFromAuthToken: string }).emailFromAuthToken =
+          "not an email";
+        next();
+      }
+    );
+
+    await agent.patch("/api/v2/folder/1").expect(400);
+  });
+
+  test("expect displayDate is updated", async () => {
+    await agent
+      .patch("/api/v2/folder/1")
+      .send({ displayDate: "2024-09-26T15:09:52" })
+      .expect(200);
+
+    const result = await db.query(
+      "SELECT to_char(displaydt, 'YYYY-MM-DD HH24:MI:SS') as displaydt FROM folder WHERE folderid = :folderId",
+      {
+        folderId: 1,
+      }
+    );
+
+    expect(result.rows[0]).toEqual({ displaydt: "2024-09-26 15:09:52" });
+  });
+
+  test("expect displayDate is updated when set to null", async () => {
+    await agent
+      .patch("/api/v2/folder/1")
+      .send({ displayDate: null })
+      .expect(200);
+
+    const result = await db.query(
+      "SELECT displaydt FROM folder WHERE folderid = :folderId",
+      {
+        folderId: 1,
+      }
+    );
+
+    expect(result.rows[0]).toStrictEqual({ displaydt: null });
+  });
+
+  test("expect 200 if user has share rights", async () => {
+    (verifyUserAuthentication as jest.Mock).mockImplementation(
+      async (
+        req: Request<
+          unknown,
+          unknown,
+          { userSubjectFromAuthToken?: string; emailFromAuthToken?: string }
+        >,
+        __,
+        next: NextFunction
+      ) => {
+        req.body.emailFromAuthToken = "test+1@permanent.org";
+        req.body.userSubjectFromAuthToken =
+          "b5461dc2-1eb0-450e-b710-fef7b2cafe1e";
+        next();
+      }
+    );
+
+    jest
+      .spyOn(folderAccess, "getAccessByFolder")
```

Comment:
We should either not mock this, or add unit tests for access.ts, because it is currently not covered by tests.

Reply:
Was able to not mock it in some tests. Others still need mocking to to cover other scenarios unavalilable with the curent fixtures. I would say changing fixtures to not mock this at all will be a bit difficult.

How can we see the test code coverage?

Reply:
You can add the `--coverage` flag to the test command in `packages/api/package.json`. The cases we're missing right now are the case where this is an `access` entry for the folder, and the case where the database call fails.